pkg/networkdevice/pinger/pinger_linux.go

```diff
@@ -41,21 +46,41 @@ func (p *LinuxPinger) Ping(host string) (*Result, error) {
 		return RunPing(&p.cfg, host)
 	}
 
-	tu, err := net.GetRemoteSystemProbeUtil(
-		pkgconfigsetup.SystemProbe().GetString("system_probe_config.sysprobe_socket"))
+	return getPing(p.sysprobeClient, clientID, host, p.cfg.Count, p.cfg.Interval, p.cfg.Timeout)
+}
+
+func getPing(client *http.Client, clientID string, host string, count int, interval time.Duration, timeout time.Duration) (*Result, error) {
+	url := sysprobeclient.ModuleURL(sysconfig.PingModule, fmt.Sprintf("/ping/%s?client_id=%s&count=%d&interval=%d&timeout=%d", host, clientID, count, interval, timeout))
+	req, err := http.NewRequest("GET", url, nil)
```

Comment:
do we want to add a timeout to this request as well:
`http.NewRequestWithContext`

Reply:
I'm not making functional changes to the individual requests. If those teams want to do that in future PRs, they are welcome to.

Reply:
There are some default timeouts on the transport as well.